openfl/component/envoy/envoy.py

```diff
@@ -80,14 +85,35 @@ def run(self):
                 logger.exception(f'Failed to get experiment: {exc}')
                 time.sleep(DEFAULT_RETRY_TIMEOUT_IN_SECONDS)
                 continue
+            
             data_file_path = self._save_data_stream_to_file(data_stream)
-            self.is_experiment_running = True
+            
             try:
                 with ExperimentWorkspace(
                         self.name + '_' + experiment_name,
                         data_file_path,
                         is_install_requirements=True
                 ):
+                    if self.review_experiment:
+                        # envoy to review the experiment before starting
+                        echo('Please review the plan before proceeding...')
+                        time.sleep(3)
+                        echo('################ PLAN START ################')
+                        with open_file('plan/plan.yaml', 'r') as f:
+                            echo(f.read())
+                        echo('################ PLAN END ################')
+                        if confirm('Do you want to accept the experiment?'):
+                            echo('Experiment accepted by Collaborator!')
+                        else:
+                            raise Exception('Experiment rejected by Collaborator')
+                            self.director_client.set_experiment_failed(
+                                experiment_name,
+                                error_code=2,
+                                error_description='Collaborator rejected the experiment'
+                            )
```

Comment:
This user interaction code should be placed in a separate module. The first idea that came to my mind - put it somewhere in cli package and import this check from there